src/Findologic/Config/FindologicConfigService.php

```diff
@@ -241,11 +262,22 @@ private function load(string $salesChannelId, string $languageId): array
     private function buildConfig(FinSearchConfigCollection $configs): array
     {
         $findologicConfig = [];
+        // Set the configuration schema for enabling inheritance
+        foreach (self::CONFIG_KEYS as $configKey) {
+            $findologicConfig[$configKey] = null;
+        }
+
         foreach ($configs as $config) {
-            $keyExists = array_key_exists($config->getConfigurationKey(), $findologicConfig);
-            if (!$keyExists || !Utils::isEmpty($config->getConfigurationValue())) {
-                $findologicConfig[$config->getConfigurationKey()] = $config->getConfigurationValue();
+            $value = $config->getConfigurationValue();
+            $key = $config->getConfigurationKey();
+            $inheritedValuePresent = array_key_exists($key, $findologicConfig);
```

Comment:
Isn't this always returning `false`? In the foreach above you explicitly set all values to `null`.